src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1032,6 +1045,91 @@ class WorkspaceService(protected val userInfo: UserInfo, dataSource: DataSource,
+              val memberGroups = group.subGroups.map(g => containerDAO.authDAO.loadGroup(g, txn).get.groupEmail.value)
+              RequestComplete(StatusCodes.OK, (memberUsers ++ memberGroups).toSeq)
+          }
+        }
+      }
+    }
+  }
+
+  def createGroup(groupRef: RawlsGroupRef) = {
+    asAdmin {
+      dataSource.inFutureTransaction() { txn =>
+        Future {
+          containerDAO.authDAO.loadGroup(RawlsGroupRef(groupRef.groupName), txn) match {
+            case Some(_) => RequestComplete(ErrorReport(StatusCodes.Conflict, s"Group ${groupRef.groupName} already exists"))
+            case None =>
+              gcsDAO.createGoogleGroup(groupRef.groupName.value)
+              //break email creation out into new function
+              containerDAO.authDAO.createGroup(RawlsGroup(groupRef.groupName, RawlsGroupEmail(gcsDAO.toGroupName(groupRef.groupName.value)), Set.empty[RawlsUserRef], Set.empty[RawlsGroupRef]), txn)
+              RequestComplete(StatusCodes.Created)
+          }
+        }
+      }
+    }
+  }
+
+  def deleteGroup(groupName: String) = {
+    asAdmin {
+      dataSource.inFutureTransaction() { txn =>
+        Future {
+          containerDAO.authDAO.loadGroup(RawlsGroupRef(RawlsGroupName(groupName)), txn) match {
+            case None => RequestComplete(ErrorReport(StatusCodes.NotFound, s"Group ${groupName} does not exist"))
+            case Some(_) =>
+              gcsDAO.deleteGoogleGroup(groupName)
+              containerDAO.authDAO.deleteGroup(RawlsGroupRef(RawlsGroupName(groupName)), txn)
+              RequestComplete(StatusCodes.OK)
+          }
+        }
+      }
+    }
+  }
+
+  def addGroupMember(groupName: String, memberEmail: String) = {
+    asAdmin {
+      dataSource.inFutureTransaction() { txn =>
+        Future {
+          containerDAO.authDAO.loadGroup(RawlsGroupRef(RawlsGroupName(groupName)), txn) match {
+            case None => RequestComplete(ErrorReport(StatusCodes.NotFound, s"Group ${groupName} does not exist"))
+            case Some(group) =>
+              gcsDAO.addMemberToGoogleGroup(groupName, memberEmail)
+              try {
+                containerDAO.authDAO.addMemberToGroup(group, memberEmail, txn)
+                RequestComplete(StatusCodes.OK)
+              } catch {
+                case re: RawlsException => RequestComplete(ErrorReport(StatusCodes.NotFound, s"Member ${memberEmail} does not exist"))
+              }
+          }
+        }
+      }
+    }
+  }
+
+  def removeGroupMember(groupName: String, memberEmail: String) = {
+    asAdmin {
+      dataSource.inFutureTransaction() { txn =>
+        Future {
+          containerDAO.authDAO.loadGroup(RawlsGroupRef(RawlsGroupName(groupName)), txn) match {
+            case None => RequestComplete(ErrorReport(StatusCodes.NotFound, s"Group ${groupName} does not exist"))
+            case Some(group) =>
+              gcsDAO.removeMemberFromGoogleGroup(groupName, memberEmail)
```

Comment:
i think per doug's comment you should be passing in `Either[RawlsUser, RawlsGroup]` to the googly dao and letting it decide how to handle them